dms-application/src/main/kotlin/team/aliens/dms/domain/remain/usecase/QueryRemainOptionsUseCase.kt

```diff
@@ -0,0 +1,42 @@
+package team.aliens.dms.domain.remain.usecase
+
+import team.aliens.dms.common.annotation.ReadOnlyUseCase
+import team.aliens.dms.domain.remain.dto.QueryRemainOptionsResponse
+import team.aliens.dms.domain.remain.dto.QueryRemainOptionsResponse.RemainOptionElement
+import team.aliens.dms.domain.remain.spi.QueryRemainOptionPort
+import team.aliens.dms.domain.remain.spi.QueryRemainStatusPort
+import team.aliens.dms.domain.remain.spi.RemainQueryUserPort
+import team.aliens.dms.domain.remain.spi.RemainSecurityPort
+import team.aliens.dms.domain.user.exception.UserNotFoundException
+
+@ReadOnlyUseCase
+class QueryRemainOptionsUseCase(
+    private val securityPort: RemainSecurityPort,
+    private val queryUserPort: RemainQueryUserPort,
+    private val queryRemainOptionPort: QueryRemainOptionPort,
+    private val queryRemainStatusPort: QueryRemainStatusPort
+) {
+
+    fun execute(): QueryRemainOptionsResponse {
+        val currentUserId = securityPort.getCurrentUserId()
+        val currentUser = queryUserPort.queryUserById(currentUserId) ?: throw UserNotFoundException
+
+        val remainOptions = queryRemainOptionPort.queryAllRemainOptionsBySchoolId(currentUser.schoolId)
+            .map {
+                RemainOptionElement(
+                    id = it.id,
+                    title = it.title,
+                    description = it.description
+                )
+            }
+
+        val remainStatus = queryRemainStatusPort.queryRemainStatusById(currentUserId)
+
+        return QueryRemainOptionsResponse(
+            selectedOption = remainStatus?.let {
+                queryRemainOptionPort.queryRemainOptionById(it.remainOptionId)?.title
```

Comment:
p1; 여기서 쿼리 날리지 말고 아래처럼 구현하는게 좋을 것 같아요.

```suggestion
remainOptions.find { it.id == remainStatus.remainOptionId }?.title
```

Reply:
넵